llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1207,25 +1207,36 @@ LogicalResult TopkOp::verify() {
   if (inputValuesType.getElementType() != outputValuesType.getElementType()) {
     return op->emitOpError("expected input/output value types to be identical");
   }
-  // Indices must be int
-  auto inputIndicesType = indices().getType().cast<ShapedType>();
+  // Indices must be int if provided
   auto outputIndicesType = outputIndices().getType().cast<ShapedType>();
-  if (!inputIndicesType.getElementType().isInteger(32) ||
-      !outputIndicesType.getElementType().isInteger(32)) {
-    return op->emitOpError("expected input/output indices types to be int");
+  if (auto inputIndices = indices()) {
+    auto inputIndicesType = inputIndices->getType().cast<ShapedType>();
+    if (!inputIndicesType.getElementType().isInteger(32) ||
+        !outputIndicesType.getElementType().isInteger(32)) {
+      return op->emitOpError("expected input/output indices types to be int");
```

Comment:
Should signal the bit width because we explicitly check it. It's confusing to users if they are using i64 types.

```suggestion
      return op->emitOpError("expected input/output indices types to be int32");
```